src/commcare_cloud/ansible/bootstrap_python.yml

```diff
@@ -0,0 +1,26 @@
+---
+- hosts: all
+  gather_facts: no
+  tasks:
+
+    - name: Check for Python 3.10
+      raw: test -e /usr/bin/python3.10
+      register: check_python
+      check_mode: no
+      failed_when: false
+      changed_when: false
+
+    - name: Install Python 3.10
+      raw: add-apt-repository -y ppa:deadsnakes/ppa && apt update && apt-get -y install python3.10
+      when: check_python.rc != 0
+
+    - name: Check symlink is setup
+      raw: python3 -c "import platform; print(platform.python_version())"
+      register: check_symlink
+      check_mode: no
+      failed_when: false
+      changed_when: false
+
+    - name: Create symlink for 3.10
+      raw: ln -sf /usr/bin/python3.10 /usr/bin/python3
```

Comment:
Is this supported on Ubuntu? My guess is it's not.

For some OSes it is unsupported and strongly discouraged to change the system Python installation in any way because scripts that are built in to the OS may not work on the new version of Python.

Here's [one example](https://cloudbytes.dev/snippets/upgrade-python-to-latest-version-on-ubuntu-linux#step-3-set-python-310-as-default) (Ubuntu) and [here's another](https://superuser.com/a/770697/86694) (MacOS).

It's fine to install new versions of Python, but I think we will need to refer to them by either reference them via full version paths like `/usr/bin/python3.10` (best option) or by putting symlinks in places where the system will not reference them (like `/usr/local/bin`).

Reply:
That's a good point. It sounds like a mixture of using this playbook to install python 3.10 with root privileges, and then using `init.sh` so decide which version of python to use would be the ideal solution then? How does that sound? And @skodde feel free to add any thoughts you may have.

Reply:
[`ansible_python_interpreter`](https://docs.ansible.com/ansible/latest/reference_appendices/interpreter_discovery.html) can be used to tell ansible to use a non-default python on target nodes.